tests/testthat/test-lancelot-multistrain.R

```diff
@@ -3515,3 +3516,154 @@ test_that("rotate strain uses correct variables", {
+          mean(y$ihr_strain[2, ], na.rm = TRUE),
+        mean(y$hfr_strain[1, ], na.rm = TRUE) >
+          mean(y$hfr_strain[2, ], na.rm = TRUE))
+  }
+
+  # Cross-immunity protects against strain 1
+  p <- lancelot_parameters(0, "england",
+                           strain_transmission = c(1, 1),
+                           rel_susceptibility = c(1, 1),
+                           cross_immunity = c(1, 0))
+  expect_equal(2, helper(p))
+
+  # Cross-immunity protects against strain 2
+  p <- lancelot_parameters(0, "england",
+                           strain_transmission = c(1, 1),
+                           rel_susceptibility = c(1, 1),
+                           cross_immunity = c(0, 1))
+  expect_equal(1, helper(p))
+
+  # Vaccine protects against both strains transmission
+  p <- lancelot_parameters(0, "england",
+                           strain_transmission = c(1, 1),
+                           rel_susceptibility = c(0, 0))
+  expect_equal(0, helper(p))
+
+  # Vaccine protects against strain 1 transmission
+  p <- lancelot_parameters(0, "england",
+                           strain_transmission = c(1, 1),
+                           rel_susceptibility = c(1, 0))
+  expect_equal(3, helper(p))
+
+  # Vaccine protects against strain 2 transmission
+  p <- lancelot_parameters(0, "england",
+                           strain_transmission = c(1, 1),
+                           rel_susceptibility = c(0, 1))
+  expect_equal(0, helper(p))
+
+  # Vaccine protects against strain 1 hospitalisation
+  p <- lancelot_parameters(0, "england",
+                           strain_transmission = c(1, 1),
+                           rel_infectivity = c(1, 1),
+                           rel_susceptibility = c(1, 1),
+                           rel_p_hosp_if_sympt = c(0, 1))
+  expect_equal(0, helper(p))
+
+  # Vaccine protects against strain 2 hospitalisation
+  p <- lancelot_parameters(0, "england",
+                           strain_transmission = c(1, 1),
+                           rel_infectivity = c(1, 1),
+                           rel_susceptibility = c(1, 1),
+                           rel_p_hosp_if_sympt = c(1, 0))
+  expect_equal(3, helper(p))
+
+  # Vaccine protects against strain 1 death
+  p <- lancelot_parameters(0, "england",
+                           strain_transmission = c(1, 1),
+                           rel_infectivity = c(1, 1),
+                           rel_susceptibility = c(1, 1),
+                           rel_p_hosp_if_sympt = c(1, 1),
+                           rel_p_death = c(0, 1))
+  expect_equal(1, helper(p))
+
+  # Vaccine protects against strain 2 death
+  p <- lancelot_parameters(0, "england",
+                           strain_transmission = c(1, 1),
+                           rel_infectivity = c(1, 1),
+                           rel_susceptibility = c(1, 1),
+                           rel_p_hosp_if_sympt = c(1, 1),
+                           rel_p_death = c(1, 0))
+  expect_equal(3, helper(p))
```

Comment:
Not sure the logic of these tests is clear to me - why sometimes 1, why sometimes 3 etc? I assume you are testing three things - why not tests those three things individually rather than just the sum of the three?

Reply:
This is now changed